core/src/main/java/org/jboss/jandex/Indexer.java

```diff
@@ -397,47 +437,128 @@ private void processRecordComponents(DataInputStream data) throws IOException {
 
     private void processAttributes(DataInputStream data, AnnotationTarget target) throws IOException {
         int numAttrs = data.readUnsignedShort();
+
         for (int a = 0; a < numAttrs; a++) {
             int index = data.readUnsignedShort();
             long attributeLen = data.readInt() & 0xFFFFFFFFL;
             byte annotationAttribute = constantPoolAnnoAttrributes[index - 1];
-            if (annotationAttribute == HAS_RUNTIME_ANNOTATION) {
-                processAnnotations(data, target);
-            } else if (annotationAttribute == HAS_RUNTIME_PARAM_ANNOTATION) {
+
+            if (!processAttribute(data, target, annotationAttribute)) {
+                skipFully(data, attributeLen);
+            }
+        }
+    }
+
+    private boolean processAttribute(DataInputStream data, AnnotationTarget target, byte annotationAttribute)
+            throws IOException {
+        boolean processed = false;
+
+        switch (annotationAttribute) {
+            case HAS_RUNTIME_INVISIBLE_ANNOTATIONS:
+            case HAS_RUNTIME_ANNOTATION:
+                processAnnotations(data, target, annotationAttribute == HAS_RUNTIME_ANNOTATION);
+                processed = true;
+                break;
+
+            case HAS_RUNTIME_INVISIBLE_PARAM_ANNOTATIONS:
+            case HAS_RUNTIME_PARAM_ANNOTATION:
                 if (!(target instanceof MethodInfo))
                     throw new IllegalStateException("RuntimeVisibleParameterAnnotations appeared on a non-method");
+
                 int numParameters = data.readUnsignedByte();
                 for (short p = 0; p < numParameters; p++) {
-                    processAnnotations(data, new MethodParameterInfo((MethodInfo) target, p));
+                    processAnnotations(data, new MethodParameterInfo((MethodInfo) target, p),
+                            annotationAttribute == HAS_RUNTIME_PARAM_ANNOTATION);
                 }
-            } else if (annotationAttribute == HAS_RUNTIME_TYPE_ANNOTATION) {
-                processTypeAnnotations(data, target);
-            } else if (annotationAttribute == HAS_SIGNATURE) {
+                processed = true;
+                break;
+
+            case HAS_RUNTIME_INVISIBLE_TYPE_ANNOTATIONS:
+            case HAS_RUNTIME_TYPE_ANNOTATION:
+                processTypeAnnotations(data, target, annotationAttribute == HAS_RUNTIME_TYPE_ANNOTATION);
+                processed = true;
+                break;
+
+            case HAS_SIGNATURE:
                 processSignature(data, target);
-            } else if (annotationAttribute == HAS_EXCEPTIONS && target instanceof MethodInfo) {
-                processExceptions(data, (MethodInfo) target);
-            } else if (annotationAttribute == HAS_INNER_CLASSES && target instanceof ClassInfo) {
-                processInnerClasses(data, (ClassInfo) target);
```

Comment:
To be fair, I find the old style (sequence of `if`s) much better to read than the new style (`switch` + `if` in each branch). The `processAttribute` method returning a `boolean` (and caller doing `skipFully` if `false`) is also not really necessary. The old style had `skipFully` as the last `else` branch, which is IMHO easier to understand than this.

Reply:
I mainly did the switch to avoid duplicating the logic for the invisible annotation cases (the method param annotation case in particular). I'm fine to put it back to the other style if it's more readable for others, however. Let me know.

Reply:
It should be straightforward to have an `if` condition matching more than 1 `annotationAttribute` value, I think?

Reply:
No question about that :-). I'll update the PR later today.

Reply:
Done